datarepo-clienttests/src/main/java/common/utils/LogsUtils.java

```diff
@@ -0,0 +1,121 @@
+package common.utils;
+
+import com.google.api.gax.core.FixedCredentialsProvider;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.logging.v2.LoggingClient;
+import com.google.cloud.logging.v2.LoggingSettings;
+import com.google.logging.v2.ListLogEntriesRequest;
+import com.google.logging.v2.LogEntry;
+import com.google.logging.v2.ProjectName;
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.TimeZone;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import runner.config.ServerSpecification;
+
+public class LogsUtils {
+  private static final Logger logger = LoggerFactory.getLogger(LogsUtils.class);
+
+  private static int minimumTimeRangeSizeInSeconds = 300; // test run has to last at least 5 minutes
+
+  private LogsUtils() {}
+
+  /**
+   * Build a Google Logging client object with application default credentials. The client object is
+   * newly created on each call to this method; it is not cached.
+   */
+  public static LoggingClient getClient() throws IOException {
+    // LoggingOptions options = LoggingOptions.getDefaultInstance(); // v1 client api
+
+    GoogleCredentials applicationDefaultCredentials =
+        AuthenticationUtils.getApplicationDefaultCredential();
+    LoggingSettings loggingServiceSettings =
+        LoggingSettings.newBuilder()
+            .setCredentialsProvider(FixedCredentialsProvider.create(applicationDefaultCredentials))
+            .build();
+    LoggingClient loggingServiceClient = LoggingClient.create(loggingServiceSettings);
+    return loggingServiceClient;
+  }
+
+  /** Download the raw logging data points. */
+  public static List<LogEntry> downloadLogEntryDataPoints(ProjectName project, String filter)
+      throws IOException {
+    LoggingClient loggingServiceClient = getClient();
+
+    // Page<LogEntry> entries = loggingClient.listLogEntries(
+    // Logging.EntryListOption.filter(filter)); // v1 client api
```

Comment:
I hope we don't have to. I used the v2 version of the API, but I think that's in beta for logging. In case it changes and breaks something, I also included the code for the v1 version of the API to fall back on, just so we don't have to go re-figure out the syntax again.